From: Jiri Kosina <jkosina@suse.cz>
Subject: Revert module.c change from -stable update
References: bnc#600364

This reverts one patche from 2.6.32.12 -stable update, as it causes
kernel crash on ia64. The other patch has been reverted in 2.6.32.14

The following two 2.6.32.12 -stable patches are reverted:

	[169/197] modules: fix incorrect percpu usage
	[171/197] module: fix __module_ref_addr()

They were incorrectly applied to 2.6.32-stable, though they should be applied
only to 2.6.33+ stable (on 2.6.32 ia64, static and dynamic percpu areas are
separate).

Signed-off-by: Jiri Kosina <jkosina@suse.cz>

---
 kernel/module.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/module.c
+++ b/kernel/module.c
@@ -571,7 +571,7 @@ static void percpu_modcopy(void *pcpudes
 	int cpu;
 
 	for_each_possible_cpu(cpu)
-		memcpy(per_cpu_ptr(pcpudest, cpu), from, size);
+		memcpy(pcpudest + per_cpu_offset(cpu), from, size);
 }
 
 #else /* ... !CONFIG_SMP */
